app/code/Magento/Config/Console/Command/ConfigSet/ProcessorFacade.php

```diff
@@ -96,13 +97,13 @@ public function __construct(
      * @param string $scope The configuration scope (default, website, or store)
      * @param string $scopeCode The scope code
      * @param boolean $lock The lock flag
+     * @param string $lockTarget
      * @return string Processor response message
      * @throws ValidatorException If some validation is wrong
-     * @throws CouldNotSaveException If cannot save config value
-     * @throws ConfigurationMismatchException If processor can not be instantiated
+     * @internal param bool $share The share flag
      * @since 100.2.0
      */
-    public function process($path, $value, $scope, $scopeCode, $lock)
+    public function process($path, $value, $scope, $scopeCode, $lock, $lockTarget = ConfigFilePool::APP_ENV)
```

Comment:
Unfortunately this backward incompatible change. Please follow the instruction here http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/#adding-parameters-in-public-methods

Reply:
@okorshenko Can you explain to me why adding an optional parameter at the end should be backwards incompatible? I'd like to understand that.

Reply:
Hi @avstudnitz this class has annotation `@api`. Adding new parameter to the method will break SPI clients of this class

Reply:
Changed. If you have a better name than `processWithLockTarget` for the new method please tell me.